internal/providers/packet/packet.go

```diff
@@ -48,3 +62,102 @@ func FetchConfig(f resource.Fetcher) (types.Config, report.Report, error) {
 
 	return util.ParseConfig(f.Logger, data)
 }
+
+// PostStatus posts a message that will show on the Packet Instance Timeline
+func PostStatus(f resource.Fetcher, errMsg error) error {
+
+	f.Logger.Info("POST message to Packet Timeline")
+	// fetch JSON from https://metadata.packet.net/metadata
+	data, err := f.FetchToBuffer(metadataUrl, resource.FetchOptions{
+		Headers: nil,
+	})
+	if data == nil {
+		return err
+	}
+
+	if err != nil {
+		return err
+	}
+
+	metadata := struct {
+		PhoneHomeURL string `json:"phone_home_url"`
+	}{}
+
+	err = json.Unmarshal(data, &metadata)
+	if err != nil {
+		return err
+
+	}
+
+	phonehomeURL := metadata.PhoneHomeURL
+	// to get phonehome IPv4
+	phonehomeURL = strings.TrimSuffix(phonehomeURL, "/phone-home")
+
+	// POST Message to phonehome IP
+	postMessageURL := phonehomeURL + "/events"
+
+	if errMsg != nil {
+
+		err = postMessage(errMsg, f, "Ignition error: "+errMsg.Error(), postMessageURL)
+		if err != nil {
+			return err
+		}
+
+	} else {
+
+		err = postMessage(errMsg, f, "Ignition: valid config", postMessageURL)
+		if err != nil {
+			return err
+		}
+
+	}
+	return err
+}
+
+// postMessage makes a post request with the supplied message to the url
+func postMessage(e error, f resource.Fetcher, message string, url string) error {
+
+	type mStruct struct {
+		State   string `json:"state"`
+		Message string `json:"message"`
+	}
+
+	m := mStruct{}
+
+	if e != nil {
```

Comment:
This if block seems to be doing the same thing regardless of the value of `e` (which also doesn't seem to be used).